bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/handler/MideaACHandler.java

```diff
@@ -0,0 +1,1501 @@
+     * Before starting 30 second delays. (More severe Wifi issue)
+     * It is reset after a successful connection
+     */
+    public void resetRetry() {
+        retry = true;
+    }
+
+    /**
+     * Limit logging of INFO connection messages to
+     * only when the device was Offline in its prior
+     * state
+     */
+    public void resetConnectionMessage() {
+        connectionMessage = true;
+    }
+
+    private ThingStatusDetail getDetail() {
+        return thing.getStatusInfo().getStatusDetail();
+    }
+
+    /**
+     * Sets Cloud Provider
+     * 
+     * @param cloudProvider Cloud Provider
+     */
+    public void setCloudProvider(CloudProvider cloudProvider) {
+        this.cloudProvider = cloudProvider;
+    }
+
+    /**
+     * Security methods
+     * 
+     * @param security security class
+     */
+    public void setSecurity(Security security) {
+        this.security = security;
+    }
+
+    /**
+     * The {@link ConnectionManager} class is responsible for managing the state of the TCP connection to the
+     * indoor AC unit evaporator.
+     *
+     * @author Jacek Dobrowolski - Initial Contribution
+     * 
+     * @author Bob Eckhoff - Revised logic to reconnect with security before each poll or command
+     * 
+     *         This gets around the issue that any command needs to be within 30 seconds of the authorization
+     *         in testing this only adds 50 ms, but allows polls at longer intervals
+     */
+    public class ConnectionManager {
+        private Logger logger = LoggerFactory.getLogger(ConnectionManager.class);
+
+        private boolean deviceIsConnected;
+        private int droppedCommands = 0;
+
+        private @Nullable Socket socket;
+        private @Nullable InputStream inputStream;
+        private @Nullable DataOutputStream writer;
+
+        private @Nullable ScheduledFuture<?> connectionMonitorJob;
+
+        private byte[] data = HexFormat.of().parseHex("C00042667F7F003C0000046066000000000000000000F9ECDB");
+
+        private String responseType = "query";
+
+        private byte bodyType = (byte) 0xc0;
+
+        private Response lastResponse = new Response(data, getVersion(), responseType, bodyType);
+        private MideaACHandler mideaACHandler;
+
+        /**
+         * Gets last response
+         * 
+         * @return byte array of last response
+         */
+        public Response getLastResponse() {
+            return this.lastResponse;
+        }
+
+        Runnable connectionMonitorRunnable = () -> {
+            logger.debug("Connecting to {} at IP {} for Poll", thing.getUID(), ipAddress);
+            disconnect();
+            connect();
+        };
+
+        /**
+         * Set the parameters for the connection manager
+         * 
+         * @param ipv4Address IP4 Address
+         * @param mideaACHandler mideaACHandler class
+         */
+        public ConnectionManager(String ipv4Address, MideaACHandler mideaACHandler) {
```

Comment:
This parameter does not seem to be used.
```suggestion
        public ConnectionManager(MideaACHandler mideaACHandler) {
```

Reply:
Done